src/pointer/ptr.rs

```diff
@@ -454,6 +487,59 @@ mod _conversions {
         }
     }
 
+    impl<'a, T, I> Ptr<'a, T, I>
+    where
+        T: 'a + ?Sized,
+        I: Invariants,
+        I::Aliasing: invariant::at_least::Shared,
+    {
+        pub(crate) fn reborrow(&'a self) -> Self {
```

Comment:
This is probably unsound. In particular, if `I::Aliasing = Exclusive`, then `Self` is an exclusive reference that can be passed by-value to methods such as `as_mut`.